scipy/stats/_stats_py.py

```diff
@@ -7705,18 +7724,32 @@ def ks_1samp(x, cdf, args=(), alternative='two-sided', method='auto'):
     x = np.sort(x)
     cdfvals = cdf(x, *args)
 
+    # TODO: add logic to find the argmax
+
     if alternative == 'greater':
-        Dplus = _compute_dplus(cdfvals)
-        return KstestResult(Dplus, distributions.ksone.sf(Dplus, N))
+        Dplus, d_location = _compute_dplus(cdfvals)
+        return KstestResult(Dplus, distributions.ksone.sf(Dplus, N),
+                            statistic_location=d_location,
+                            statistic_sign='')
 
     if alternative == 'less':
-        Dminus = _compute_dminus(cdfvals)
-        return KstestResult(Dminus, distributions.ksone.sf(Dminus, N))
+        Dminus, d_location = _compute_dminus(cdfvals)
+        return KstestResult(Dminus, distributions.ksone.sf(Dminus, N),
+                            statistic_location=d_location,
+                            statistic_sign='')
 
     # alternative == 'two-sided':
-    Dplus = _compute_dplus(cdfvals)
-    Dminus = _compute_dminus(cdfvals)
-    D = np.max([Dplus, Dminus])
+    Dplus, dplus_location = _compute_dplus(cdfvals)
+    Dminus, dminus_location = _compute_dminus(cdfvals)
+    if Dplus > Dminus:
+        D = Dplus
+        d_location = dplus_location
+        d_sign = '+'
```

Comment:
What do you think about indicating the sign with `1` and `-1` (number) instead of a string? 
Also, even though it's redundant with the specified `alternative`, I'd suggest including it in the result for the one-sided alternative.

Reply:
Sure